src/test/java/com/microsoft/sqlserver/jdbc/resiliency/BasicConnectionTest.java

```diff
@@ -268,6 +271,47 @@ public void testPooledConnectionLang() throws SQLException {
         }
     }
 
+    /**
+     * Test PreparedStatement Cache is cleared after reconnect
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testPreparedStatementCacheShouldBeCleared() throws SQLException {
+        try (SQLServerConnection con = (SQLServerConnection) ResiliencyUtils.getConnection(connectionString)) {
+            int cacheSize = 2;
+            String query = String.format("/*testPreparedStatementCacheShouldBeCleared_%s*/SELECT 1; -- ",

```

Comment:
I would test a statement that produces multiple result sets:

`select 1;select 1`

Reply:
Scanning through the list of recent fixes reminded me that we should also cover exception cases. ICR should still work afterwards. Examples:

`select 1/0`
`select 1;select 1/0` <- Processing all result sets and only processing the first set.